pkg/storage/allocator_test.go

```diff
@@ -2046,6 +2058,108 @@ func TestAllocatorComputeAction(t *testing.T) {
 	}
 }
 
+// TestAllocatorComputeActionDisableStatsRebalance is used to verify whether the cluster could balance out
+// if we disable stats-based-rebalance.
+func TestAllocatorRebalanceTargetDisableStatsRebalance(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	ctx := context.Background()
+	stopper, g, _, a, _ := createTestAllocator( /* deterministic */ false)
+	defer stopper.Stop(ctx)
+
+	st := a.storePool.st
+	EnableStatsBasedRebalancing.Override(&st.SV, true)
+	// Make sure the cluster's initial status is not balanced.
+	// store 1, 2, 3 has 200 replicas, store 4 has only 30 replicas.
+	// So obviously, if we want the replica is homogeneous within each store,
+	// we should rebalance some replicas from store 1, 2, 3 to store 4.
+	// However, we set the WritesPerSecond of store 4 is much bigger than other store,
+	// so if we enable stats-based rebalance, we couldn't spread out the replicas on store 1, 2, 3.
+	stores := []*roachpb.StoreDescriptor{
+		{
+			StoreID: 1,
+			Node:    roachpb.NodeDescriptor{NodeID: 1},
+			Capacity: roachpb.StoreCapacity{
+				RangeCount:      200,
+				WritesPerSecond: 30,
+			},
+		},
+		{
+			StoreID: 2,
+			Node:    roachpb.NodeDescriptor{NodeID: 2},
+			Capacity: roachpb.StoreCapacity{
+				RangeCount:      200,
+				WritesPerSecond: 30,
+			},
+		},
+		{
+			StoreID: 3,
+			Node:    roachpb.NodeDescriptor{NodeID: 3},
+			Capacity: roachpb.StoreCapacity{
+				RangeCount:      200,
+				WritesPerSecond: 30,
+			},
+		},
+		{
+			StoreID: 4,
+			Node:    roachpb.NodeDescriptor{NodeID: 4},
+			Capacity: roachpb.StoreCapacity{
+				RangeCount:      30,
+				WritesPerSecond: 100,
+			},
+		},
+	}
+	sg := gossiputil.NewStoreGossiper(g)
+	sg.GossipStores(stores, t)
+
+	desc := roachpb.RangeDescriptor{
+		RangeID: firstRange,
+		Replicas: []roachpb.ReplicaDescriptor{
+			{
+				StoreID:   1,
+				NodeID:    1,
+				ReplicaID: 1,
+			},
+			{
+				StoreID:   2,
+				NodeID:    2,
+				ReplicaID: 2,
+			},
+			{
+				StoreID:   3,
+				NodeID:    3,
+				ReplicaID: 3,
+			},
+		},
+	}
+	for i := 0; i < 50; i++ {
+		target, _ := a.RebalanceTarget(
+			context.Background(),
+			config.Constraints{},
+			nil,
+			testRangeInfo(desc.Replicas, desc.RangeID),
+			storeFilterThrottled,
+			false,
+		)
+		if target != nil {
+			t.Errorf("expected no balance, but got %d", target.StoreID)
+		}
+	}
+
+	for i := 0; i < 50; i++ {
+		target, _ := a.RebalanceTarget(
+			context.Background(),
+			config.Constraints{},
+			nil,
+			testRangeInfo(desc.Replicas, desc.RangeID),
+			storeFilterThrottled,
+			true,
```

Comment:
Going by the recent addition to our style guide (https://github.com/cockroachdb/cockroach/pull/19185), this should be `true /* disableStatsBasedRebalance */`

Reply:
Done now. Thanks!